### PR TITLE
Loosen Trident backendType check

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
```yaml
parameters:
  backendType: ontap-nas-economy
```
Was spotted in the wild.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/storagecapabilities/storagecapabilities.go [hunk 1]
```diff
@@ -147,9 +148,9 @@ var storageClassToProvisionerKeyMapper = map[string]func(sc *storagev1.StorageCl
 	"csi.trident.netapp.io": func(sc *storagev1.StorageClass) string {
 		//https://netapp-trident.readthedocs.io/en/stable-v20.04/kubernetes/concepts/objects.html#kubernetes-storageclass-objects
 		val := sc.Parameters["backendType"]
-		if val == "ontap-nas" {
+		if strings.HasPrefix(val, "ontap-nas") {
```
Comment (ShellyKa13) [inline pkg/storagecapabilities/storagecapabilities.go:151]: perhaps we want to do check for contains? instread of prefix?

File: pkg/storagecapabilities/storagecapabilities.go [hunk 2]
```diff
@@ -147,9 +148,9 @@ var storageClassToProvisionerKeyMapper = map[string]func(sc *storagev1.StorageCl
...
 			return "csi.trident.netapp.io/ontap-nas"
-		} else if val == "ontap-san" {
+		} else if strings.HasPrefix(val, "ontap-san") {
```
[the remark this answers is not in the document]
Comment (akalenyu) [reply pkg/storagecapabilities/storagecapabilities.go:153]: should change to a bunch of `if`s?

[the remark this answers is not in the document]
Comment (ShellyKa13) [reply pkg/storagecapabilities/storagecapabilities.go:153]: its only 2 ifs:
if strings.HasPrefix(val, "ontap-nas") {
    return "csi.trident.netapp.io/ontap-nas
}
if strings.HasPrefix(val, "ontap-san") {
    return "csi.trident.netapp.io/ontap-san"
}

[the remark this answers is not in the document]
Comment (akalenyu) [reply pkg/storagecapabilities/storagecapabilities.go]: done

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm

File: pkg/storagecapabilities/storagecapabilities.go (hunk 1 above)
Comment (akalenyu) [reply pkg/storagecapabilities/storagecapabilities.go:151]: Yeah I am hesitant to start supporting all the different ones, so kept this strictly about ontap-nas/ontap-san error we saw

**Pull request merged**